tuf/ngclient/updater.py

```diff
@@ -122,12 +126,38 @@ def refresh(self) -> None:
             RepositoryError: Metadata failed to verify in some way
             DownloadError: Download of a metadata file failed in some way
         """
+        #                                     json.load()
+        # Load self.spec_version using self._load_local_metadata() here from disk
+        # Exception would be raised if spec_version is not found locally. Set to "1" in that case
+        repository_versions = self._get_repository_versions()
+
+        # Updating self.spec_version
+        specver_message = _get_spec_version(
+            repository_versions, self.spec_version, SUPPORTED_VERSIONS
+        )
+
+        if specver_message(1):
+            logger.warning(specver_message(1))
+        self.spec_version = specver_message(0)
+        self._persist_metadata("spec_version", json.dumps(self.spec_version))
 
         self._load_root()
         self._load_timestamp()
         self._load_snapshot()
         self._load_targets(Targets.type, Root.type)
 
+    def _get_repository_versions(self) -> List[str]:
+        """Returns a list of all the repository versions."""
+
+        url = f"{self._metadata_base_url}supported-versions.json"
+
+        with self._fetcher.download_file(
```

Comment:
If this fails, we probably want to return `["1"]`

Reply:
Or `[""]`

Reply:
Also, should I use `download_bytes()` instead because afai can see, `download_bytes()` just uses `download_file()` inside of `fetcher.py`

Reply:
Oh, good call—do that!